### ISSUE TITLE
testing: add runIfMain

### ISSUE THREAD

**Issue by ry** [denoland/deno_std#152]
Once we support `imports.meta.url` and have a way for testing if the current script is the main script, we want to have a two line snippit of code that will allow any test to be executed as a standalone test `deno foo_test.ts` rather than running the main entry point.
Essentially something like this:
https://github.com/tensorflow/tensorflow/blob/c4e61ac1fd0b3b48461e2876b20fe56c09faa2ff/tensorflow/python/lib/io/tf_record_test.py#L484

The most basic form would be something like this:
```ts
if (args[1] == imports.meta.url) {
  testMain();
}
``` 
But perhaps we can make this shorter by having a function like `runIfMain(imports.meta.url)`

This came up in: https://github.com/denoland/deno_std/pull/129

**Comment by ry** [comment]
import.meta.url is now available in 0.2.10 so runTestsIfMain should be doable now.

**Comment by bartlomieju** [comment]
I guess this belongs to `testing/mod.ts` but that file contains this notice: `// Do not add imports in this file in order to be compatible with Node.`. So `import { args } from "deno";` isn't desired there. Any thoughts?

**Comment by bartlomieju** [comment]
@ry any more thoughts on this?

**Comment by ry** [comment]
@bartlomieju Yes - I think if we implement `window.location` as described in https://github.com/denoland/deno/issues/1750 then we can do
```js
if (imports.meta.url == window.location.toString()) {
  runTests();
}
```

**Comment by ry** [comment]
FYI `window.location` [has landed](https://github.com/denoland/deno/commit/42408febe8cdf9e30ff8d1a3bb13f4994906c53b) in Deno v0.3.0

One can test if the script is main by using this:
```ts
if (window.location.toString() == import.meta.url) {
  console.log("main");
}
```
We should be able to adapt for `testing.runIfMain(import.meta)`

**Comment by bartlomieju** [comment]
Closed in #215

**Comment by bartlomieju** [comment]
CC @ry
